### ISSUE TITLE
Client is not disconnected when the JWT is expired

### ISSUE THREAD

**Issue by jitbasemartin** [emqx/emqx#8799]
### What happened?

The client is not disconnected and continue to receive data when the JWT is expired

### What did you expect to happen?

Should not receive and send data after the JWT expiration

### How can we reproduce it (as minimally and precisely as possible)?

I configured EMQX to auth with JWT, then I created a JWT with expiration in 2 minutes. 
I can subscribe to a topic to the MQTT broker (v5), receive data.
After 2 minutes (expiration of the JWT) I continue to receive data

### EMQX version

```
sysdescr  : EMQX
version   : 5.0.6
datetime  : 2022-08-24T19:48:45.906626340+00:00
uptime    : 3 hours, 43 minutes, 9 seconds
```

### OS version

Use Docker mac

**Comment by zmstone** [comment]
some related discussion here: https://github.com/emqx/emqx/discussions/9067

**Comment by dnlwgnd** [comment]
Hi, 
I was also surprised by the current behavior. I worked without ACL first and expected a disconnect upon expiration, whicht did not happen.

I tried to add ACL rules in my JWT Token as suggested and observed this:
When the token gets expired and a publish action occurs, EMXQ will log a violation and does not deliver the message
```
2022-11-14T12:01:16.801004+00:00 [warning] clientid: mqttx_1ebd2b68_1668425469000, line: 655, mfa: emqx_channel:process_publish/2, msg: cannot_publish_to_topic, peername: 10.23.85.192:443, reason: not_authorized, topic: test
```
However, the log does not indicate an expired token as the true reason.
The client is NOT disconnected and also receives NO error. This disconcerts me and seems inconsistent. From a publishing clients perspective it fails silently and might not be noticed.

I would like to see the situation improved.
Kind Regards

**Comment by jitbasemartin** [comment]
Always presents in v5.0.24 :(

**Comment by MacherelR** [comment]
Hi, Anyone found a solution to counter this problem ? It seems still the case that even if JWT expiration time is reached the connection is not stopped (if subscribed)... 
I was wondering maybe I'm using it the wrong way by having long lasting connections but as I should listen to a command topic I can't really find another way, has anyone found a solution for this ?

**Comment by zmstone** [comment]
While the "disconnect client if token expired" feature is still missing, I do want to make an update about logging improvements.

later v5 (currently 5.5.0) releases have logging improved to indicate token being expired.
e.g. if you trace a client which tries to login with an expired token, there should be a trace like:
```
2024-02-02T17:21:52.068403+00:00 [AUTHN] aaa@127.0.0.1:33398 msg: invalid_jwt_claims, claims: {<<"exp">>,1706894451}, jwt: .......
```

**Comment by zmstone** [comment]
For publishing clients. One can configure `deny_action=disconnect` to force the client to disconnect (and hopefully acquire a new token and reconnect).

For subscribing clients, it's a bit awkward, we will do something about it, but not sure when it will get prioritized.

**Comment by id** [comment]
Released in 5.7.0 https://github.com/emqx/emqx/pull/12947
